src/app/_services/editor-pref.service.spec.ts

```diff
@@ -0,0 +1,62 @@
+import { QueryList } from '@angular/core';
+import { async, TestBed } from '@angular/core/testing';
+import { CodemirrorComponent } from 'ng2-codemirror';
+
+import { EditorPrefService } from '.';
+
+function makeQueryList(): QueryList<CodemirrorComponent> {
+  // tslint:disable: no-any
+  return ([
+    {
+      instance: {
+        setOption(name: string, value: string): void {
+          console.log('set ' + name + ' to ' + value);
+        },
+      },
+    },
+  ] as any) as QueryList<CodemirrorComponent>;
+}
+
+describe('editor pref service', () => {
+  let service: EditorPrefService;
+
+  beforeEach(async(() => {
+    localStorage.removeItem('editor-pref');
+    TestBed.configureTestingModule({
+      providers: [EditorPrefService],
+    }).compileComponents();
+    service = TestBed.get(EditorPrefService);
+  }));
+
+  afterEach(() => {
+    localStorage.removeItem('editor-pref');
+  });
+
+  it('should use the default theme by default', () => {
+    const theme = service.getEditorPref();
+    expect(theme).toEqual('default');
+  });
+
+  it('should report on whether the current theme is the default theme', () => {
+    expect(service.currentThemeIsDefault()).toEqual(true);
+    service.toggleTheme(makeQueryList());
+    expect(service.currentThemeIsDefault()).toEqual(false);
+  });
+
+  it('can toggle the current theme', () => {
+    expect(service.getEditorPref()).toEqual('default');
+    service.toggleTheme(makeQueryList());
+    expect(service.getEditorPref()).not.toEqual('default');
+  });
```

Comment:
here you check if the `setOptions` is called (if you created a spy for that):

    const queryList = makeQueryList();
    service.toggleTheme(queryList);
    expect(queryList[0].instance.setOption).toHaveBeenCalledWith('theme', 'midnight');